pkg/i2gw/providers/gce/README.md

```diff
@@ -0,0 +1,41 @@
+# GCE Provider
+
+The project supports translating ingress-gce specific annotations.
+
+## Implementation-specific features
+
+The following implementation-specific features are supported:
+
+- There is no corresponding Gateway HTTP path type that matches ingress `ImplementationSpecific` path that contains `*`, except for `/*` which will map to `/` prefix. Please consider switch to `Prefix`, `Exact`, or `ImplementationSpecific` path without `*` before converting Ingress to Gateway.
+- For `ImplementationSpecific` paths without `*`, they directly map to `Exact` path type.
+
+| ingress `ImplementationSpecific` Path | map to Gateway Path                    |
+| ------------------------------------- | -------------------------------------- |
+| /*                                    | / Prefix                               |
+| /v1                                   | /v1 Exact                              |
+| /v1/                                  | /v1/ Exact                             |
+| /v1/*                                 | /v1 Prefix but not /v1(slash required) |
+
+## Feature list
+Currently supported:
+- [Basic Internal Ingress](https://github.com/GoogleCloudPlatform/gke-networking-recipes/tree/main/ingress/single-cluster/ingress-internal-basic)
+- [Basic external Ingress](https://github.com/GoogleCloudPlatform/gke-networking-recipes/tree/main/ingress/single-cluster/ingress-external-basic)
+- [Ingress with custom default backend](https://github.com/GoogleCloudPlatform/gke-networking-recipes/tree/main/ingress/single-cluster/ingress-custom-default-backend)
+
+Not supported:
+ - [HTTP-to-HTTPS redirect](https://github.com/GoogleCloudPlatform/gke-networking-recipes/tree/main/ingress/single-cluster/ingress-https)
+ 
+   - In GKE Ingress, Ingress using [HTTP-to-HTTPS redirect](https://cloud.google.com/kubernetes-engine/docs/how-to/ingress-configuration#https_redirect) is provisioned with FrontendConfig, so it is only [supported by External Ingress][https://cloud.google.com/kubernetes-engine/docs/how-to/ingress-configuration#associating_frontendconfig_with_your_ingress] only.
+   However, this feature is [not supported](https://cloud.google.com/kubernetes-engine/docs/how-to/deploying-gateways#configure_http-to-https_redirects) in the corresponding GKE GatewayClass `gke-l7-gxlb`.
+
+TBD(Mostly BackendConfig compatiblity):
+ - [Ingress with custom HTTP health check](https://github.com/GoogleCloudPlatform/gke-networking-recipes/tree/main/ingress/single-cluster/ingress-custom-http-health-check)
```

Comment:
I think this should map to [HealthCheckPolicy](https://cloud.google.com/kubernetes-engine/docs/how-to/configure-gateway-resources#configure_health_check) - of course that would require translating Ingress extensions to Gateway extensions, I'm not sure if we've done that before. @LiorLieberman and @mlavacca what do you think about this idea?

Reply:
I think its a good idea. There is also an open issue for that - https://github.com/kubernetes-sigs/ingress2gateway/issues/125. 

It will require some work in the framework to allow outputting gateway extensions but this is a good direction imo.

@arkodg and I were chatting at KubeCon about defining some kind of IR(intermediate representation) and then support two different kind of providers - input_providers, those will implement the logic of (Ingress/custom configurations --> IR) and output_providers, those will read the IR and transform to Gateway and gateway extensions. This still requires some more refinement though.

I think for this PR it is ok to leave it as TODO and implement the extensions as a followup.
@sawsa307 - happy to chat more about gateway extensions if you are interested.

Reply:
Thanks a lot for the reference and discussion. I will try to add support for these use case by the order of the list. @LiorLieberman I'll definitely schedule a meeting with you when I'm working in the follow up that creates gateway extension and custom http health check!